controllers/account/ec2.go

```diff
@@ -377,18 +387,18 @@ func deleteFedrampInitializationResources(reqLogger logr.Logger, client awsclien
 }
 
 // retrieveVpcs list of all VPCs with appropriate tag
-func retrieveVpcs(reqLogger logr.Logger, client awsclient.Client, region string) (*ec2.DescribeVpcsOutput, bool, error) {
-	cleaned := false
+func retrieveVpcs(reqLogger logr.Logger, client awsclient.Client, region string) (*ec2.DescribeVpcsOutput, error) {
 	// Make dry run to certify required authentication
 	_, err := client.DescribeVpcs(&ec2.DescribeVpcsInput{
 		DryRun: aws.Bool(true),
 	})
 
 	// If we receive AuthFailure, do not attempt to clean resources
-	if aerr, ok := err.(awserr.Error); ok {
+	var aerr awserr.Error
+	if errors.As(err, &aerr) {
```

Comment:
Is there a specific reason that this is used over the previous code paradigm?

Reply:
Primarily has to do with being able to assert error types even when errors are wrapped when using things like `fmt.Errorf("some error %w", err)`. https://go.dev/blog/go1.13-errors